src/codelabs/async-await.md

```diff
@@ -321,16 +369,39 @@ synchronous code before the first `await` keyword executes immediately.
 Run the following example to see how execution proceeds within an `async`
 function body. What do you think the output will be?
 
-<iframe
-  src="{{site.dartpad-embed}}?id=d7abfdea1ae5596e96c7c0203d975dba"
-  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 40px"
-  frameborder="no"
-  height="600"
-  width="100%">
-</iframe>
+<?code-excerpt "async_await/bin/get_order.dart (with-counter)"?>
+```dart:run-dartpad:height-530px
+Future<String> createOrderMessage() async {
```

Comment:
This code is different from the previous code. @legalcodes are you OK with that?
[Old code](https://dart.dev/codelabs/async-await#example-execution-within-async-functions)
[New code](https://dart-dev-staging-1.firebaseapp.com/codelabs/async-await#example-execution-within-async-functions)

Reply:
Let's keep the function signatures as they were -- @galeyang recommended these based on user feedback (i.e. these were carefully chosen for pedagogical reasons). The rest of the changes look fine (reformatting, removing unneeded import).

Reply:
Here's part of the old code that is relevant to this discussion:

```
void createOrderMessage () async {
  print('Awaiting user order...');
  var order = await getUserOrder();
  print('Your order is: $order');
}

Future<String> getUserOrder() {
  // Imagine that this function is more complex and slow.
  return Future.delayed(Duration(seconds: 4), () => 'Large Latte');
}

main() async {
  countSeconds(4);
  await createOrderMessage();
}
```

Here's the reasoning behind the changes I made, along with comments about what I think is peculiar about the original code:

 1. At the start of (original) [Working with futures: async and await](https://dart.dev/codelabs/async-await#working-with-futures-async-and-await) we write:

    > If [an async] ... function doesn’t explicitly return a value, then the return type is `Future<void>`:
    >
    > ```dart
    > Future<void> main() async {
    > ```

    So, ...
    1. **Shouldn't we be following our own guidelines?**
    2. Why do we given an example of attributing a return type of `Future<void>` to `main()` if we never do so in the codelab?

 2. Pedagogically, I find it strange that as we work through the codelab, `String createOrderMessage()` starts off with a **semantically meaningful name** and consistent return type. Then later we repurpose the function to merely _print the order message_ (rather than return it) and so change the signature to `void createOrderMessage()`. By doing so we're introducing an incongruity. One solution would be to change the function name to `printOrderMessage()`.

3. Regardless of our choice of function name in the previous point (say, `void printOrderMessage()`), by using a return type of `void` we're again not following our own guideline (the guideline I mention in point 1), and we're not explaining why we're deviating from our own guideline.

Thoughts?

Reply:
I'm reverting to the original code (modulo dartfmt changes, since the original code was manually formatted, and a change in duration for the delay). I'll move the comment above to an issue if you think that it is worth pursuing.

Reply:
Btw, when `createOrderMessage()` is declared as void, we get the following issue reported by the analyzer:

```nocode
'await' applied to 'void', which is not a 'Future'. dart(await_only_futures)
```

Reply:
I've added an `ignore` directive for now.
The code has been reverted.

Reply:
I created #2128 to track this.

Reply:
Thanks for moving it to a new issue. We have some relevant unresolved discussion earlier. I will take a look and continue the discussion here. Thanks!